server/app/views/admin/programs/ProgramBlockPredicateConfigureView.java

```diff
@@ -0,0 +1,621 @@
+                          groupCount.incrementAndGet(),
+                          Optional.of(andNode))));
+    } else {
+      valueRowContainer.with(
+          renderValueRow(questionDefinitions, 1, /* maybeAndNode= */ Optional.empty()));
+    }
+
+    formTag.with(valueRowContainer);
+    formTag.with(div(makeSvgTextButton("Add values", Icons.ADD).withId("predicate-add-value-row")));
+    formTag.with(submitButton("Save condition").withClasses("my-4"));
+
+    return formTag;
+  }
+
+  private static DivTag renderConfiguratorEligibilityHeader() {
+    return div(
+            text("Applicant is eligible when:"),
+            input()
+                .withType("hidden")
+                .withName("predicateAction")
+                .withValue(PredicateAction.ELIGIBLE_BLOCK.name()))
+        .withClasses("font-bold");
+  }
+
+  private static DivTag renderConfiguratorVisibilityHeader(
+      Optional<PredicateDefinition> maybeExistingPredicate) {
+    DivTag container =
+        div("Screen is ").withClasses("font-bold", ReferenceClasses.PREDICATE_ACTION);
+    SelectTag selectTag = select().withName("predicateAction");
+
+    ImmutableList.of(PredicateAction.HIDE_BLOCK, PredicateAction.SHOW_BLOCK)
+        .forEach(
+            action ->
+                selectTag.with(
+                    option(action.toDisplayString())
+                        .withClasses(
+                            ReferenceClasses.MULTI_OPTION_QUESTION_OPTION,
+                            ReferenceClasses.MULTI_OPTION_VALUE)
+                        .withCondSelected(
+                            maybeExistingPredicate
+                                .map(PredicateDefinition::action)
+                                .map(existingAction -> existingAction.equals(action))
+                                .orElse(false))
+                        .withValue(action.name())));
+
+    container.with(selectTag);
+
+    return container;
+  }
+
+  private ImmutableList<PredicateExpressionNode> getExistingAndNodes(
+      PredicateDefinition existingPredicate) {
+    switch (existingPredicate.computePredicateFormat()) {
+      case SINGLE_QUESTION:
+        {
+          return ImmutableList.of(
+              PredicateExpressionNode.create(
+                  AndNode.create(ImmutableList.of(existingPredicate.rootNode()))));
+        }
+
+      case OR_OF_SINGLE_LAYER_ANDS:
+        {
+          return existingPredicate.rootNode().getOrNode().children();
+        }
+
+      default:
+        {
+          throw new RuntimeException(
+              String.format(
+                  "Unrecognized predicate format: %s", existingPredicate.computePredicateFormat()));
+        }
+    }
+  }
+
+  private DivTag renderQuestionHeaders(
+      ImmutableList<QuestionDefinition> questionDefinitions,
+      Optional<PredicateDefinition> maybeExistingPredicate) {
+    DivTag container = div().withClasses("flex", "py-4");
+    int columnNumber = 1;
```

Comment:
Look at the next comment first, but if you keep this... Define this in each if, it's not used after them so it shouldn't errorpronely bleed between them.

Reply:
done, though IntelliJ doesn't like it
![image](https://user-images.githubusercontent.com/473671/210879383-04cafc21-190a-4ba2-ad57-d065bf0962b4.png)